internal/beater/beater.go

```diff
@@ -924,12 +925,23 @@ func queryClusterUUID(ctx context.Context, esClient *elasticsearch.Client) error
 	return nil
 }
 
-type nopProcessingSupporter struct{}
+type nopProcessingSupporter struct {
+	processors []beat.Processor
+}
 
-func (nopProcessingSupporter) Close() error {
+func (*nopProcessingSupporter) Close() error {
 	return nil
 }
 
-func (nopProcessingSupporter) Create(cfg beat.ProcessingConfig, _ bool) (beat.Processor, error) {
+func (p *nopProcessingSupporter) Processors() []string {
+	procList := []string{}
+	for _, proc := range p.processors {
+		procList = append(procList, proc.String())
+	}
+	return procList
+}
+
+func (p *nopProcessingSupporter) Create(cfg beat.ProcessingConfig, _ bool) (beat.Processor, error) {
+	p.processors = append(p.processors, cfg.Processor)
```

Comment:
AFAICS the libbeat default implementation of Create does not store `cfg.Processor` on the supporter. I don't think we should either - that might be a source of concurrency issues. Can we remove this append, and have `Processors()` just return an empty slice? AFAICS `Processors()` isn't relevant to apm-server anyway; it's only used by `cmd/instance/beat.go` in elastic/beats, and we don't use that code.

Reply:
Thank you!, good point! :+1: